### ISSUE TITLE
Allow option to change private key from default id_rsa

### ISSUE THREAD

**Issue by xori** [schollz/gojot#24]
[master] ~ sdees
sdees has capability to SSH tunnel to a remote host in order to
keep documents synced across devices.
Would you like to set this up? (y/n) y
Enter remote address (default: localhost): XXXXXXX
Enter remote user (default: XXXXXXX): XXX
Enter remote port (default: 22):
Which editor do you want to use: vim (default), nano, or emacs?
Enter new document name (default: notes.txt):
Pulling from remote...
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x28 pc=0x53fcb3]

goroutine 1 [running]:
panic(0x7a3200, 0xc04200c090)
        /usr/local/go/src/runtime/panic.go:500 +0x1af
github.com/urfave/cli.HandleAction.func1(0xc04209f8c8)
        /home/zack/work/src/github.com/urfave/cli/app.go:476 +0x2a5
panic(0x7a3200, 0xc04200c090)
        /usr/local/go/src/runtime/panic.go:458 +0x251
golang.org/x/crypto/ssh.(_connection).clientAuthenticate(0xc042138080, 0xc04212c0c0, 0x0, 0xa)
        /home/zack/work/src/golang.org/x/crypto/ssh/client_auth.go:52 +0x443
golang.org/x/crypto/ssh.(_connection).clientHandshake(0xc042138080, 0xc0420f41d0, 0xd, 0xc04212c0c0, 0x0, 0x0)
        /home/zack/work/src/golang.org/x/crypto/ssh/client.go:107 +0x2f4
golang.org/x/crypto/ssh.NewClientConn(0x9b1640, 0xc042118018, 0xc0420f41d0, 0xd, 0xc04209edb0, 0x9b1640, 0xc042118018, 0x0, 0x0, 0xc0420f41d0, ...)
        /home/zack/work/src/golang.org/x/crypto/ssh/client.go:75 +0x10c
golang.org/x/crypto/ssh.Dial(0x8098b3, 0x3, 0xc0420f41d0, 0xd, 0xc04209edb0, 0x80956b, 0x2, 0xc0420f41d0)
        /home/zack/work/src/golang.org/x/crypto/ssh/client.go:171 +0xba
main.syncDown()
        /home/zack/Documents/sdees/sync.go:44 +0x3a6
main.run()
        /home/zack/Documents/sdees/run.go:112 +0xf14
main.main.func2(0xc042060500, 0x0, 0x0)
        /home/zack/Documents/sdees/main.go:319 +0x7d8
reflect.Value.call(0x7903e0, 0x83d1a8, 0x13, 0x809a0c, 0x4, 0xc04209f868, 0x1, 0x1, 0x50b6bf, 0x7ff8e0, ...)
        /usr/local/go/src/reflect/value.go:434 +0x5cf
reflect.Value.Call(0x7903e0, 0x83d1a8, 0x13, 0xc04209f868, 0x1, 0x1, 0x80ba15, 0x1, 0x8094f0)
        /usr/local/go/src/reflect/value.go:302 +0xab
github.com/urfave/cli.HandleAction(0x7903e0, 0x83d1a8, 0xc042060500, 0x0, 0x0)
        /home/zack/work/src/github.com/urfave/cli/app.go:485 +0x1e7
github.com/urfave/cli.(*App).Run(0xc0420da000, 0xc04203e230, 0x1, 0x1, 0x0, 0x0)
        /home/zack/work/src/github.com/urfave/cli/app.go:244 +0x5a2
main.main()
        /home/zack/Documents/sdees/main.go:391 +0xcb2

**Comment by xori** [comment]
The same crash happens on subsequent runs of `sdees`

**Comment by schollz** [comment]
What is your remote host OS?

Also does it work locally? (Try running `sdees --local`)

**Comment by xori** [comment]
My remote host is a Ubuntu 16.04.1 box. 

And `sdees --local` does work
```
 ~ sdees --local
Enter password for editing 'notes.txt':
Enter password again:
Wrote 3rZ7mZ0.Vlo673.xWRaVRD.gpg.
+1 words.
```

**Comment by schollz** [comment]
Okay I'm a little stumped - I can't seem to replicate it.

I thought maybe it had something to do with creating a new directory, but I just tested on a clean remote dir with the latest Windows release with no problem. My local is Windows 10 and my remote is Ubuntu 14.04.05. 

I can try later with Ubuntu 16 as remote.

**Comment by xori** [comment]
I found the issue! I use multiple SSH keys and the server I was connecting to doesn't use my default `id_rsa` key. I had hoped that `sdees` would use my `~/.ssh/config` file for determining the connection parameters. 

Something you will want to add is a good error message when your private key doesn't work. To avoid this from happening. And *maybe* a config option to give the name of the private key inside your `~/.ssh/` [instead of defaulting to `id_rsa`](https://github.com/schollz/sdees/blob/dd8a9efe5075d842d36ca29eaf20298806b49b58/main.go#L190). Or better yet, [using the system's ssh configs.](https://github.com/moul/advanced-ssh-config)

But I got my `sdees` working now, I don't know if you want me to leave the issue open to track that error message "feature" so I'll leave it to you to close if you want.
